src/io/perun.clj

```diff
@@ -116,6 +116,54 @@
       (pm/set-meta fileset updated-files)
       (commit fileset tmp))))
 
+(defn meta-by-ext
+  [fileset file-exts]
+  (->> fileset
+       boot/user-files
+       (boot/by-ext file-exts)
+       (keep pm/meta-from-file)))
+
+(defn content-pre-wrap
+  "Wrapper for input parsing tasks. Calls `parse-form` on new or changed
+  files with extensions in `file-exts`, adds `tracer` to `:io.perun/trace`
+  and writes html files for subsequent tasks to process, if desired. Pass
+  `pod` if one is needed for parsing"
+  [parse-form file-exts tracer & [pod]]
+  (let [tmp     (boot/tmp-dir!)
+        prev-fs (atom nil)]
+    (boot/with-pre-wrap fileset
+      (let [changed-files (->> (boot/fileset-diff @prev-fs fileset :hash)
+                               boot/user-files
+                               (boot/by-ext file-exts)
+                               add-filedata)
+            changed-meta (trace tracer
+                                (if pod
+                                  (pod/with-call-in @pod ~(parse-form changed-files))
+                                  (eval (parse-form changed-files))))
+            input-fs (-> (if @prev-fs
+                           (pm/set-meta fileset (meta-by-ext @prev-fs file-exts))
+                           fileset)
+                         (pm/set-meta changed-meta))
+            input-meta (meta-by-ext input-fs file-exts)
+            output-meta (doall
+                         (for [{:keys [path parsed filename] :as entry*} input-meta]
+                           (let [page-filepath (string/replace path #"(?i).[a-z]+$" ".html")
```

Comment:
Extensions can contain nearly any character including '-' and '_'.  Perhaps this should be constructed by the file-exts parameter?